crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -494,10 +494,20 @@ pub struct PipelineCache {
 }
 
 impl PipelineCache {
+    /// Returns an iterator over the pipelines in the pipeline cache.
     pub fn pipelines(&self) -> impl Iterator<Item = &CachedPipeline> {
         self.pipelines.iter()
     }
 
+    /// Returns a vector of all currently waiting pipelines IDs.
+    pub fn waiting_pipelines(&self) -> Vec<CachedPipelineId> {
+        let mut waiting_pipelines_vec: Vec<CachedPipelineId> = Vec::new();
+        self.waiting_pipelines
+            .iter()
+            .for_each(|id| waiting_pipelines_vec.push(*id));
+        waiting_pipelines_vec
```

Comment:
```suggestion
    pub fn waiting_pipelines(&self) -> impl Iterator<Item = CachedPipelineId> {
        self.waiting_pipelines.iter().copied()
```
Let's try not to allocate memory when accessing this.

Reply:
Apologies, wasn't aware of the `copied` method. Still not 100% on the difference between clone and copy and wasn't sure what was the best way to do this. Thanks for showing me this! I will research this topic further and take this into consideration in the future.